bundles/org.openhab.binding.teslascope/src/main/java/org/openhab/binding/teslascope/internal/TeslascopeWebTargets.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.teslascope.internal;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles performing the actual HTTP requests for communicating with the TeslascopeAPI.
+ *
+ * @author Paul Smedley - Initial Contribution
+ *
+ */
+@NonNullByDefault
+public class TeslascopeWebTargets {
+    private static final int TIMEOUT_MS = 30000;
+    private static final String BASE_URI = "https://teslascope.com/api/vehicle/";
+    private final Logger logger = LoggerFactory.getLogger(TeslascopeWebTargets.class);
+    private HttpClient httpClient;
+
+    public TeslascopeWebTargets(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    public String getDetailedInformation(String publicID, String apiKey)
+            throws TeslascopeCommunicationException, TeslascopeAuthenticationException {
+        String getDetailedInformationUri = BASE_URI + publicID + "/detailed?api_key=" + apiKey;
+        String response = invoke(getDetailedInformationUri);
+        logger.trace("Received response: \"{}\"", response);
+        return response;
+    }
+
+    public void sendCommand(String publicID, String apiKey, String command)
+            throws TeslascopeCommunicationException, TeslascopeAuthenticationException {
+        String sendCommandUri = BASE_URI + publicID + "/command/" + command + "?api_key=" + apiKey;
+        String response = invoke(sendCommandUri);
+        logger.trace("Received response: \"{}\"", response);
+        return;
+    }
+
+    private String invoke(String uri) throws TeslascopeCommunicationException, TeslascopeAuthenticationException {
+        logger.debug("Calling url: {}", uri);
+        String jsonResponse = "";
+        int status = 0;
+        try {
+            Request request = httpClient.newRequest(uri).method(HttpMethod.GET)
+                    .header("Content-Type", "application/json").timeout(TIMEOUT_MS, TimeUnit.MILLISECONDS);
```

Comment:
```suggestion
                    .header(HttpHeader.CONTENT_TYPE, HttpHeaders.Values.APPLICATION_JSON).timeout(TIMEOUT_MS, TimeUnit.MILLISECONDS);
```

Reply:
`Content-Type` should not be set for `GET` requests, since there is no body. Strict servers might even reject such requests.